src/App.tsx

```diff
@@ -1,34 +1,62 @@
 import { useKeycloak } from "@react-keycloak/web";
-import { Route, Routes, Navigate } from 'react-router-dom';
+import { Route, Routes, Navigate } from "react-router-dom";
 import { useEffect, useState } from "react";
-import { api } from './api/api';
-import Home from './pages/Home';
-import CaretakerForm from './pages/CaretakerForm';
-import LoginPage from './pages/LoginPage';
-import store from './store/RootStore';
+import { api } from "./api/api";
+import Home from "./pages/Home";
+import CaretakerForm from "./pages/CaretakerForm";
+import LoginPage from "./pages/LoginPage";
+import store from "./store/RootStore";
 import CaretakerSearch from "./pages/CaretakerSearch";
+import ProfileSelection from "./pages/ProfileSelection";
 
 function App() {
   const { keycloak, initialized } = useKeycloak();
+  const [isXsrfTokenFetched, setIsXsrfTokenFetched] = useState(false);
+  const [isUserDataFetched, setIsUserDataFetched] = useState(false);
   const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
     const fetchXsrfToken = async () => {
       if (keycloak.authenticated && !store.user.xsrfToken) {
         await api.getXsrfToken();
+        setIsXsrfTokenFetched(true);
+      }
+    };
```

Comment:
`setIsXsrfTokenFetched(true);` musi być zawsze, a nie tylko w `if` czyli jeśli do tej pory nie było tokenu, bo jeśli ktoś tylko odświeży stronę to xsrf token będzie ustawiony ale mimo to strona się nie załaduje.
```suggestion
    const fetchXsrfToken = async () => {
      if (keycloak.authenticated && !store.user.xsrfToken) {
        await api.getXsrfToken();
      }
      setIsXsrfTokenFetched(true);
    };
```